lib/bsmOauth.ts

```diff
@@ -1,86 +1,116 @@
-import axios, { AxiosError } from 'axios';
-import BsmOauthError from './error.js';
-import ErrorType from './types/errorType.js';
-import { RawBsmOAuthResource, RawBsmOAuthToken } from './types/rawOAuthType.js';
-import { BsmStudent, BsmStudentResource } from './types/student.js';
-import { BsmTeacher, BsmTeacherResource } from './types/teacher.js';
-import UserRole from './types/userRole.js';
+import axios, { AxiosError, isAxiosError } from "axios";
+import BsmOauthError from "./error.js";
+import ErrorType from "./types/errorType.js";
+import { RawBsmOAuthResource, RawBsmOAuthToken } from "./types/rawOAuthType.js";
+import { BsmStudent, BsmStudentResource } from "./types/student.js";
+import { BsmTeacher, BsmTeacherResource } from "./types/teacher.js";
+import UserRole from "./types/userRole.js";
 
 export default class BsmOauth {
-
   constructor(clientId: string, clientSecret: string) {
     this.bsmAuthPayload = {
       clientId,
-      clientSecret
+      clientSecret,
     };
```

Comment:
constructor 내의 clientId와 clientSecret를 클래스 속성으로 만들면 페이로드 객체를 사용할 필요가 없어지지 않을까요?

```suggestion
constructor(
private readonly clientId: string, 
private readonly clientSecret: string
) {}
```